lib/index.js

```diff
@@ -1,46 +1,51 @@
 'use strict';
-var Assert = require('assert');
-var GoodSqueeze = require('good-squeeze');
-var Loggly = require('loggly');
-var Merge = require('lodash.merge');
 
-
-function GoodLoggly (events, config) {
-  if (!(this instanceof GoodLoggly)) {
-    return new GoodLoggly(events, config);
+const Assert = require('assert');
+const Stream = require('stream');
+const Loggly = require('loggly');
+const Merge = require('lodash.merge');
+
+
+class GoodLoggly extends Stream.Writable {
+  constructor (config) {
+    config = Merge({}, config);
+
+    Assert(typeof config.token === 'string', 'Loggly API token required');
+    Assert(typeof config.subdomain === 'string', 'Loggly subdomain required');
+
+    super({ objectMode: true, decodeStrings: false });
+    this._config = config;
+    this._client = Loggly.createClient({
+      token: config.token,
+      subdomain: config.subdomain,
+      json: true,
+      // Tags in JSON logs don't seem to appear properly in Loggly when
+      // sent with the X-LOGGLY-TAG header
+      useTagHeader: false,
+      tags: Array.isArray(config.tags) ? config.tags : []
+    });
   }
+  _write (event, encoding, callback) {
+    const data = Merge({}, event);
```

Comment:
I don't THINK you need to do this anymore. Each reporter pipeline get's it's own copy so you are free to edit it as you see fit.